beacon_node/network/src/beacon_processor/mod.rs

```diff
@@ -153,18 +217,92 @@ impl<E: EthSpec> WorkEvent<E> {
         aggregate: SignedAggregateAndProof<E>,
     ) -> Self {
         Self {
-            message_id,
-            peer_id,
-            work: Work::Aggregate(Box::new(aggregate)),
+            drop_during_sync: true,
+            work: Work::GossipAggregate {
+                message_id,
+                peer_id,
+                aggregate: Box::new(aggregate),
+            },
+        }
+    }
+
+    /// Create a new `Work` event for some block.
+    pub fn gossip_beacon_block(
+        message_id: MessageId,
+        peer_id: PeerId,
+        block: Box<SignedBeaconBlock<E>>,
+    ) -> Self {
+        Self {
+            drop_during_sync: true,
```

Comment:
Are we confident that sync can handle getting to the head reliably without the help of gossip blocks? I thought I'd noticed that my node sometimes got kicked out of a sync loop by a new gossip block

Reply:
Hmm let's ask @AgeManning

Reply:
Oh yeah. So there is a heuristic where we look at gossipblocks even when we are still synced. If it is close to head its valuable. 

See this condition: https://github.com/sigp/lighthouse/blob/master/beacon_node/network/src/sync/manager.rs#L453

Reply:
As discussed offline with @AgeManning, I've changed it to always process gossip blocks, even if syncing.